projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/JobsService.java

```diff
@@ -140,7 +140,9 @@ private boolean isInvocationSuccessful(Optional<WebHookResult> resultHolder) {
   public boolean updateJob(DataJob jobInfo) {
     var dataJob =
         jobsRepository.existsById(jobInfo.getName()) ? jobsRepository.save(jobInfo) : null;
-    dataJobMetrics.updateInfoGauges(dataJob);
+    if (dataJob != null) {
```

Comment:
Nice find. 

I think I would prefer if there was only a single if statement. 

like 

```
if(jobsRepository.existsById(jobInfo.getName())){
      dataJobMetrics.updateInfoGauges(jobsRepository.save(jobInfo));
      return true;
    }
    return false;
 ```
 
 
 But happy to merge this also.

Reply:
Agreed.